internal/services/search/search_service_resource.go

```diff
@@ -233,6 +245,11 @@ func resourceSearchServiceCreate(d *pluginsdk.ResourceData, meta interface{}) er
 		return fmt.Errorf("%q SKUs in %q mode can have a maximum of 3 partitions, got %d", string(services.SkuNameStandardThree), string(services.HostingModeHighDensity), partitionCount)
 	}
 
+	// NOTE: Semantic Search SKU cannot be set if the SKU is 'free'
+	if skuName == services.SkuNameFree && semanticSearchSku != string(services.SearchSemanticSearchDisabled) {
```

Comment:
which would mean we don't need to cast this one:

```suggestion
	if skuName == services.SkuNameFree && semanticSearchSku != services.SearchSemanticSearchDisabled {
```

Reply:
Fixed.